app/controllers/constraints_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class ConstraintsController < AuthenticationController
+  before_action :set_planning_application
+  before_action :ensure_constraint_edits_unlocked, only: %i[edit update]
+
+  def show; end
+
+  def edit; end
+
+  def update
+    @planning_application.constraints = params[:planning_application][:constraints].reject(&:blank?)
+    if @planning_application.save!
```

Comment:
If we use a `bang!` here we need to rescue the error I think otherwise it will just blow up and won't reach the `else` block